llama_index/readers/file/base.py

```diff
@@ -93,9 +94,13 @@ def __init__(
         if input_files:
             self.input_files = []
             for path in input_files:
+                if not os.path.isfile(path):
+                    raise ValueError(f"File {path} does not exist.")
                 input_file = Path(path)
                 self.input_files.append(input_file)
         elif input_dir:
+            if not os.path.isdir(input_dir):
```

Comment:
should we also check that the directory actually has files? maybe in `self._add_files()` ? Right now a user could still load an empty directory (which is probably not intended)

Reply:
done and added test